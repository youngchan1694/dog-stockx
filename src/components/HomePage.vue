<template>
  <v-container fluid>
    <v-data-iterator :items="dogs" items-per-page.sync="2" hide-default-footer>
      <template v-slot:default="props">
        <v-row>
          <v-col
            v-for="item in props.items"
            :key="item.name"
            cols="12"
            sm="6"
            md="4"
            lg="3"
          >
            <v-card>
              <v-card-title class="subheading font-weight-bold">{{
                item.name
              }}</v-card-title>

              <v-divider></v-divider>

              <v-list>
                <v-list-item>
                  <v-list-item-content>Breed:</v-list-item-content>
                  <v-list-item-content class="align-end">{{
                    item.breed
                  }}</v-list-item-content>
                </v-list-item>

                <v-list-item>
                  <v-list-item-content>Gender:</v-list-item-content>
                  <v-list-item-content class="align-end">{{
                    item.gender
                  }}</v-list-item-content>
                </v-list-item>

                <v-list-item>
                  <v-list-item-content>Age:</v-list-item-content>
                  <v-list-item-content class="align-end">{{
                    item.age
                  }}</v-list-item-content>
                </v-list-item>

                <v-list-item>
                  <v-list-item-content>Shots:</v-list-item-content>
                  <v-list-item-content class="align-end">
                    <span v-for="shot of item.shots" :key="shot.name">
                      {{ shot.name }}
                    </span>
                  </v-list-item-content>
                </v-list-item>

                <v-list-item>
                  <v-list-item-content>Neutered:</v-list-item-content>
                  <v-list-item-content class="align-end">{{
                    item.neutered
                  }}</v-list-item-content>
                </v-list-item>

                <v-list-item>
                  <v-list-item-content>Weight:</v-list-item-content>
                  <v-list-item-content class="align-end"
                    >{{ item.weight }} lb</v-list-item-content
                  >
                </v-list-item>
              </v-list>
              <v-img :src="getImage(item.pic)" />
            </v-card>
          </v-col>
        </v-row>
      </template>
    </v-data-iterator>
  </v-container>
</template>

<script>
import axios from "axios";

export default {
  name: "HomePage",
  data: () => ({
    dogs: []
  }),
  mounted() {
    this.getDogs();
  },
  methods: {
    async getDogs() {
      const response = await axios.post("http://localhost:4000/graphql", {
        query: `{
          dogs {
            name
            pic
            area {
              city
              state
              zip
            }
            gender
            breed
            age
            shots {
              name
              date
            }
            neutered
            weight
          }
        }`
      });
      this.dogs = response.data.data.dogs;
    },
    //Date comes back in UNIX so remember to convert
    getImage(fileName) {
      return require(`../../public/static/dogs/${fileName}`);
    }
  }
};
</script>
