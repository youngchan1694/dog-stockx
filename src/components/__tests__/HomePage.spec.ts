import Vue from "vue";
import Vuetify from "vuetify";
import { mount } from "@vue/test-utils";
import HomePage from "../HomePage.vue";
Vue.use(Vuetify);
const vuetify = new Vuetify();

describe("HomePage.vue", () => {
  const wrapper = mount(HomePage, { vuetify });

  test("matches the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
