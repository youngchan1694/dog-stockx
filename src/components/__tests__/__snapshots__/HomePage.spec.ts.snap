// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue matches the snapshot 1`] = `
"<div class=\\"container container--fluid\\">
  <div class=\\"row\\">
    <div class=\\"col\\">
      <div class=\\"container\\">
        <div class=\\"row\\">
          <div class=\\"col col-12\\">
            <div class=\\"v-input v-input--hide-details theme--light v-text-field v-text-field--solo-flat\\">
              <div class=\\"v-input__control\\">
                <div class=\\"v-input__slot\\">
                  <div class=\\"v-input__prepend-inner\\">
                    <div class=\\"v-input__icon v-input__icon--prepend-inner\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-magnify theme--light\\"></i></div>
                  </div>
                  <div class=\\"v-text-field__slot\\"><label for=\\"input-4\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Search</label><input id=\\"input-4\\" type=\\"text\\"></div>
                  <div class=\\"v-input__append-inner\\">
                    <div class=\\"v-input__icon v-input__icon--clear\\"><button disabled=\\"disabled\\" type=\\"button\\" aria-label=\\"clear icon\\" class=\\"v-icon notranslate v-icon--disabled v-icon--link mdi mdi-close theme--light\\"></button></div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"row\\">
          <div class=\\"col\\">
            <div class=\\"v-input theme--light v-text-field v-select v-select--is-multi\\">
              <div class=\\"v-input__control\\">
                <div role=\\"button\\" aria-haspopup=\\"listbox\\" aria-expanded=\\"false\\" aria-owns=\\"list-7\\" class=\\"v-input__slot\\">
                  <div class=\\"v-select__slot\\"><label for=\\"input-7\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Cities</label>
                    <div class=\\"v-select__selections\\"><input id=\\"input-7\\" readonly=\\"readonly\\" type=\\"text\\" aria-readonly=\\"false\\" autocomplete=\\"off\\"></div>
                    <div class=\\"v-input__append-inner\\">
                      <div class=\\"v-input__icon v-input__icon--append\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-menu-down theme--light\\"></i></div>
                    </div><input type=\\"hidden\\">
                  </div>
                  <div class=\\"v-menu\\">
                    <!---->
                  </div>
                </div>
                <div class=\\"v-text-field__details\\">
                  <div class=\\"v-messages theme--light\\">
                    <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class=\\"col\\">
            <div class=\\"v-input theme--light v-text-field v-select v-select--is-multi\\">
              <div class=\\"v-input__control\\">
                <div role=\\"button\\" aria-haspopup=\\"listbox\\" aria-expanded=\\"false\\" aria-owns=\\"list-12\\" class=\\"v-input__slot\\">
                  <div class=\\"v-select__slot\\"><label for=\\"input-12\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">States</label>
                    <div class=\\"v-select__selections\\"><input id=\\"input-12\\" readonly=\\"readonly\\" type=\\"text\\" aria-readonly=\\"false\\" autocomplete=\\"off\\"></div>
                    <div class=\\"v-input__append-inner\\">
                      <div class=\\"v-input__icon v-input__icon--append\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-menu-down theme--light\\"></i></div>
                    </div><input type=\\"hidden\\">
                  </div>
                  <div class=\\"v-menu\\">
                    <!---->
                  </div>
                </div>
                <div class=\\"v-text-field__details\\">
                  <div class=\\"v-messages theme--light\\">
                    <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class=\\"col\\">
            <div class=\\"v-input theme--light v-text-field v-select v-select--is-multi\\">
              <div class=\\"v-input__control\\">
                <div role=\\"button\\" aria-haspopup=\\"listbox\\" aria-expanded=\\"false\\" aria-owns=\\"list-17\\" class=\\"v-input__slot\\">
                  <div class=\\"v-select__slot\\"><label for=\\"input-17\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Zip Codes</label>
                    <div class=\\"v-select__selections\\"><input id=\\"input-17\\" readonly=\\"readonly\\" type=\\"text\\" aria-readonly=\\"false\\" autocomplete=\\"off\\"></div>
                    <div class=\\"v-input__append-inner\\">
                      <div class=\\"v-input__icon v-input__icon--append\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-menu-down theme--light\\"></i></div>
                    </div><input type=\\"hidden\\">
                  </div>
                  <div class=\\"v-menu\\">
                    <!---->
                  </div>
                </div>
                <div class=\\"v-text-field__details\\">
                  <div class=\\"v-messages theme--light\\">
                    <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class=\\"col\\">
            <div class=\\"v-input theme--light v-text-field v-select v-select--is-multi\\">
              <div class=\\"v-input__control\\">
                <div role=\\"button\\" aria-haspopup=\\"listbox\\" aria-expanded=\\"false\\" aria-owns=\\"list-22\\" class=\\"v-input__slot\\">
                  <div class=\\"v-select__slot\\"><label for=\\"input-22\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Genders</label>
                    <div class=\\"v-select__selections\\"><input id=\\"input-22\\" readonly=\\"readonly\\" type=\\"text\\" aria-readonly=\\"false\\" autocomplete=\\"off\\"></div>
                    <div class=\\"v-input__append-inner\\">
                      <div class=\\"v-input__icon v-input__icon--append\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-menu-down theme--light\\"></i></div>
                    </div><input type=\\"hidden\\">
                  </div>
                  <div class=\\"v-menu\\">
                    <!---->
                  </div>
                </div>
                <div class=\\"v-text-field__details\\">
                  <div class=\\"v-messages theme--light\\">
                    <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class=\\"col\\">
            <div class=\\"v-input theme--light v-text-field v-select v-select--is-multi\\">
              <div class=\\"v-input__control\\">
                <div role=\\"button\\" aria-haspopup=\\"listbox\\" aria-expanded=\\"false\\" aria-owns=\\"list-27\\" class=\\"v-input__slot\\">
                  <div class=\\"v-select__slot\\"><label for=\\"input-27\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Breeds</label>
                    <div class=\\"v-select__selections\\"><input id=\\"input-27\\" readonly=\\"readonly\\" type=\\"text\\" aria-readonly=\\"false\\" autocomplete=\\"off\\"></div>
                    <div class=\\"v-input__append-inner\\">
                      <div class=\\"v-input__icon v-input__icon--append\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-menu-down theme--light\\"></i></div>
                    </div><input type=\\"hidden\\">
                  </div>
                  <div class=\\"v-menu\\">
                    <!---->
                  </div>
                </div>
                <div class=\\"v-text-field__details\\">
                  <div class=\\"v-messages theme--light\\">
                    <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class=\\"col\\">
            <div class=\\"v-input theme--light v-text-field v-select v-select--is-multi\\">
              <div class=\\"v-input__control\\">
                <div role=\\"button\\" aria-haspopup=\\"listbox\\" aria-expanded=\\"false\\" aria-owns=\\"list-32\\" class=\\"v-input__slot\\">
                  <div class=\\"v-select__slot\\"><label for=\\"input-32\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Ages</label>
                    <div class=\\"v-select__selections\\"><input id=\\"input-32\\" readonly=\\"readonly\\" type=\\"text\\" aria-readonly=\\"false\\" autocomplete=\\"off\\"></div>
                    <div class=\\"v-input__append-inner\\">
                      <div class=\\"v-input__icon v-input__icon--append\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-menu-down theme--light\\"></i></div>
                    </div><input type=\\"hidden\\">
                  </div>
                  <div class=\\"v-menu\\">
                    <!---->
                  </div>
                </div>
                <div class=\\"v-text-field__details\\">
                  <div class=\\"v-messages theme--light\\">
                    <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class=\\"col\\">
            <div class=\\"v-input theme--light v-text-field v-select v-select--is-multi\\">
              <div class=\\"v-input__control\\">
                <div role=\\"button\\" aria-haspopup=\\"listbox\\" aria-expanded=\\"false\\" aria-owns=\\"list-37\\" class=\\"v-input__slot\\">
                  <div class=\\"v-select__slot\\"><label for=\\"input-37\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Neutured</label>
                    <div class=\\"v-select__selections\\"><input id=\\"input-37\\" readonly=\\"readonly\\" type=\\"text\\" aria-readonly=\\"false\\" autocomplete=\\"off\\"></div>
                    <div class=\\"v-input__append-inner\\">
                      <div class=\\"v-input__icon v-input__icon--append\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-menu-down theme--light\\"></i></div>
                    </div><input type=\\"hidden\\">
                  </div>
                  <div class=\\"v-menu\\">
                    <!---->
                  </div>
                </div>
                <div class=\\"v-text-field__details\\">
                  <div class=\\"v-messages theme--light\\">
                    <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class=\\"col\\">
            <div class=\\"v-input theme--light v-text-field v-select v-select--is-multi\\">
              <div class=\\"v-input__control\\">
                <div role=\\"button\\" aria-haspopup=\\"listbox\\" aria-expanded=\\"false\\" aria-owns=\\"list-42\\" class=\\"v-input__slot\\">
                  <div class=\\"v-select__slot\\"><label for=\\"input-42\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Weights</label>
                    <div class=\\"v-select__selections\\"><input id=\\"input-42\\" readonly=\\"readonly\\" type=\\"text\\" aria-readonly=\\"false\\" autocomplete=\\"off\\"></div>
                    <div class=\\"v-input__append-inner\\">
                      <div class=\\"v-input__icon v-input__icon--append\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-menu-down theme--light\\"></i></div>
                    </div><input type=\\"hidden\\">
                  </div>
                  <div class=\\"v-menu\\">
                    <!---->
                  </div>
                </div>
                <div class=\\"v-text-field__details\\">
                  <div class=\\"v-messages theme--light\\">
                    <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <hr role=\\"separator\\" aria-orientation=\\"horizontal\\" class=\\"v-divider theme--light\\">
  <div class=\\"row\\" style=\\"padding: 0% 2% 0% 2%;\\">
    <div class=\\"col\\">
      <div class=\\"v-data-iterator\\">
        <div>No data available</div>
      </div>
    </div>
  </div>
</div>"
`;
